iaso/api/org_units.py

```diff
@@ -74,6 +74,13 @@ class OrgUnitViewSet(viewsets.ViewSet):
     def get_queryset(self):
         return OrgUnit.objects.filter_for_user_and_app_id(self.request.user, self.request.query_params.get("app_id"))
 
+    def get_creator_name(self, creator):
+        if creator is None:
+            return None
+        if creator["first_name"] is not None or creator["last_name"] is not None:
+            return f"{creator['username']} ({creator['first_name']} {creator['last_name']})"
+        return creator["username"]
```

Comment:
We now have 3 places with almost the same function and it will be a source of inconsistencies:

1. [`iaso/iaso/api/org_units.py`](https://github.com/BLSQ/iaso/blob/9e4023c9b43056cb490659856451d58ec9d0701a/iaso/api/org_units.py#L77-L82)
2. [`iaso/models/org_unit.py`](https://github.com/BLSQ/iaso/blob/9e4023c9b43056cb490659856451d58ec9d0701a/iaso/models/org_unit.py#L247-L252)
3. [`iaso/models/base.py`](https://github.com/BLSQ/iaso/blob/9e4023c9b43056cb490659856451d58ec9d0701a/iaso/models/base.py#L1117-L1122)

Could you please factorize the `get_creator_name()` function and add unit tests for it?

E.g. this could be put in a new file, e.g. in [`iaso/utils/models/`](https://github.com/BLSQ/iaso/tree/9e4023c9b43056cb490659856451d58ec9d0701a/iaso/utils/models) and imported everywhere else:

```python
def get_creator_name(creator: User = None, username: str = "", first_name: str = "", last_name: str = "") -> str:
    if isinstance(creator, User):
        username = creator.username
        first_name = creator.first_name
        last_name = creator.last_name
    if username and first_name and last_name:
        return f"{username} ({first_name} {last_name})"
    elif username:
        return username
    return ""
```